### PR TITLE
pm: add some missing npm_ env vars

### PR DESCRIPTION
**Pull request by nektro**
Closes https://github.com/oven-sh/bun/issues/73
Closes https://github.com/oven-sh/bun/issues/2793
Closes https://github.com/oven-sh/bun/issues/13803

### CODE REVIEW

**Comment by robobun** [comment]
@nektro, your commit 52684c19b9916e14b79bbec5450ab6c1539f4588 is building: [`#5249`](https://buildkite.com/bun/bun/builds/5249)

File: src/resolver/package_json.zig [hunk 1]
```diff
@@ -1035,6 +1036,35 @@ pub const PackageJSON = struct {
                     }
                 }
             }
+            read_config: {
+                if (json.asProperty("config")) |config_prop| {
```
Comment (dylan-conway) [inline src/resolver/package_json.zig]: this looks identical to the block above, might be a handy helper something like `config_prop.e_object.createStringMap()`

**Review by dylan-conway — COMMENTED** [review]
A test should be added to `bun-workspace.test.ts` for multiple package.json with `config` fields

**Comment by hadarziv-army** [comment]
It looks like it should close my issue too #13803

**Review by dylan-conway — CHANGES_REQUESTED** [review]
Could you also add a test for `npm_command` `publish` in `bun-install-registry.test.ts`

**Review by dylan-conway — APPROVED** [review]

**Comment by whyman** [comment]
This is awesome!

**Pull request merged**